lib/update/cluster/plan.go

```diff
@@ -425,26 +425,18 @@ func shouldUpdateCoreDNS(client *kubernetes.Clientset) (bool, error) {
 	return false, nil
 }
 
-func shouldUpdateEtcd(installedRuntimeApp, updateRuntimeApp app.Application, packages pack.PackageService) (*etcdVersion, error) {
+func shouldUpdateEtcd(installedRuntimePackage, updateRuntimePackage loc.Locator, packages pack.PackageService) (*etcdVersion, error) {
 	// TODO: should somehow maintain etcd version invariant across runtime packages
-	runtimePackage, err := schema.GetDefaultRuntimePackage(installedRuntimeApp.Manifest)
-	if err != nil {
-		return nil, trace.Wrap(err, "error fetching runtime package for %v", installedRuntimeApp.Package)
-	}
 	var updateEtcd bool
-	installedVersion, err := getEtcdVersion(*runtimePackage, packages)
+	installedVersion, err := getEtcdVersion(installedRuntimePackage, packages)
 	if err != nil {
 		if !trace.IsNotFound(err) {
```

Comment:
Shouldn't "not found" also be a terminal scenario in this case? Since the etcd migrate operation is now basically copying old directory to new directory, the old one must always be present. Otherwise "installedVersion" below will be empty and we'll run into the same scenario as the customer S reported to us yesterday. Am I missing something?

Reply:
It should be the other way around - the etcd data directory migration path needs to support the case of the empty source version and fallback to a version file as it used to be.